sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -932,7 +932,19 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
       testData2.collect().toSeq.map(r => Row(r.getInt(0) ^ r.getInt(1) ^ 39)))
   }
 
+  test("SPARK-37646: lit") {
+    assert(lit($"foo") == $"foo")
```

Comment:
I don't find any unit tests like this. So just add simple tests to cover 3 cases in the code.